src/gateway/http.js

```diff
@@ -69,8 +69,34 @@ HTTP.prototype = Gateway.extends({
       assign(requestParams, httpOptions.configure(requestParams))
     }
 
+    if (httpOptions.onRequestWillStart) {
+      httpOptions.onRequestWillStart(requestParams)
+    }
+
     const httpRequest = handler
-      .request(requestParams, (httpResponse) => this.onResponse(httpResponse))
+      .request(requestParams, (httpResponse) => this.onResponse(httpResponse, httpOptions, requestParams))
+
+    httpRequest.on('socket', socket => {
+      if (httpOptions.onRequestSocketAssigned) {
+        httpOptions.onRequestSocketAssigned(requestParams)
+      }
+
+      socket.on('lookup', () => {
```

Comment:
These three socket event listeners are a bit tricky to unit test because they don't get executed when we use fauxJax to intercept the request. Ideas are welcome :)

Reply:
the only option is to mock `handler.request` and return something we control, which is a lot of work. You can also test this on the integration tests, they should work fine.